content/blog/2018/10/2018-10-15-custom-war-packager.adoc

```diff
@@ -0,0 +1,235 @@
+With such patches Jenkins starts up in few seconds with configuration-as-code and dozens of bundled plugins.
+
+So, how to build custom Jenkinsfile Runner images?
+Although there is no release so far, it is not something which can stop us as you see above.
+
+```yaml
+##...
+## WAR Configuration from above
+##...
+
+buildSettings:
+  jenkinsfileRunner:
+    source:
+      groupId: "io.jenkins"
+      artifactId: "jenkinsfile-runner"
+      build:
+        noCache: true
+      source:
+        git: https://github.com/jenkinsci/jenkinsfile-runner.git
+        commit: 8ff9b1e9a097e629c5fbffca9a3d69750097ecc4
+    docker:
+      base: "jenkins/jenkins:2.138.2"
+      tag: "onenashev/cwp-jenkinsfile-runner-demo"
+      build: true
+```
+
+You can find a Demo of Jenkinsfile Runner packaging with Custom WAR Packager
+link:https://github.com/jenkinsci/custom-war-packager/tree/master/demo/jenkinsfile-runner[here].
+
+== More info
+
+There are many other features which are not described in this blogpost.
+For example, it is possible to alter Maven build settings or to add/replace libraries within the Jenkins core (e.g. Remoting).
+Please see the link:https://github.com/jenkinsci/custom-war-packager/blob/master/README.md[Custom WAR Packager documentation] for more information.
+There is a number of demos available in the repository.
+
+If you are interested to contribute to the repository,
+please create pull requests and CC link:https://github.com/oleg-nenashev/[@oleg-nenashev]
+and link:https://github.com/raul-arabaolaza[Raul Arabaolaza] who is the second maintainer now working on Jenkins test automation flows.
+
+== What's next?
+
+There is still a many improvements which could be done in the tool in order to make it more efficient:
+
+// TODO: issue links
+
+* Add upper bounds checks for transitive plugin dependencies so that the conflicts are discovered during the build
+* Allow passing all kinds of system properties and Java options via configuration YAML
+* Improve Jenkinsfile Runner to improve performance
+* Integrate the tool into Jenkins Integration test flows
+  (see link:https://github.com/jenkins-infra/pipeline-library/blob/master/vars/essentialsTest.groovy[essentialsTest()]
+   in the Jenkins Pipeline library)
+
+Many other tasks could be implemented in Custom WAR Packager,
+but even now it is available to all Jenkins users so that they can build their own Jenkins bundles with it.
+
+== Want to know more?
+
+If you are going to link:https://www.cloudbees.com/devops-world/nice[DevOps World - Jenkins World in Nice] on Oct 22-25,
+I will be presenting Custom WAR Packager at the Community Booth during the lunch demo sessions.
+We will be also repeating our link:https://sched.co/FIox[A Cloud Native Jenkins] talk together with Carlos Sanchez where we will show how Ephemeral Jenkins works with Pluggable Storage.
+I believe that Jenkins X team is also going to present their project using Custom WAR Packager.
```

Comment:
Would be good to have this confirmed so it sounds like we all work for the same company ;)

Reply:
confirmed